### ISSUE TITLE
Backlog could store and restore message tags ?

### ISSUE THREAD

**Issue by stacyharper** [weechat/weechat#1734]
Hello there,

I'm working on some [Mattermost backend] for a while now. I recently made a change to rely more on the weechat buffer backlogs to improve performance (avoid asking multiple time the same post message to the server).

[Mattermost backend]: https://sr.ht/~reedwade/wee-matter/

My issue is new reply_to messages that need older messages to have their tags correctly setted-up. That cause problems if we close and open back Weechat frequently.

So basically, I'd like the Weechat backlogger to store and restore tags. Ideally in a retro-compatible way.

Is it correct that it is impossible to do atm ? Is the subject already been thought ? Is there a no go for this ?
I would be happy to contribute on this if you confirm it could be merged.

Thanks a lot !

**Comment by flashcode** [comment]
Hi,

This is a good question already discussed a couple of times on IRC, the answer is not so simple, it's technically possible, but was not designed to do this.

Today, the "backlog" feature is based on logs written by the logger plugin.
These logs aims to be human readable, so without tags, color codes (though it's possible to add ANSI color codes), etc.
Tags are not something you should see when just showing the file outside WeeChat, and can be annoying when scripts or external tools read files (for example to produce statistics).

So I see only 3 solutions, each one having drawbacks:

1. Keep the current behavior, ie no-go for the feature. Drawback: you'll never have tags in backlog.
2. Add another "log" file in addition to the existing one, which could be binary, compressed, with tags for example. Drawback: double storage needed.
3. Add tags in the current log files. Drawback: it will be hard to display manually or parse automatically log files.

I'm not closed to any solution, but until now, I didn't found something satisfying.

**Comment by stacyharper** [comment]
> 3. Add tags in the current log files. Drawback: it will be hard to display manually or parse automatically log files.

That was my idea. I confess I dont really see why it will be that harder to use (manually or with scripts). It will add a column before or between the two first ones. It still is splitable with a simple command : `cut -d'    ' -f2-` basically. The only thing is that just cating the log file isnt practical anymore. We could also make this optional, as we do for ansi color codes (which I use and love).
